aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,31 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer) acquires BlockResource {
+        system_addresses::assert_aptos_framework(framework_signer);
+
+        let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+
+        let genesis_id = @0x0;
+        emit_new_block_event(
+            framework_signer,
+            &mut block_metadata_ref.new_block_events,
+            NewBlockEvent {
+                hash: genesis_id,
+                epoch: reconfiguration::current_epoch(),
```

Comment:
Are we planning to trigger a reconfiguration event in the writeset as well?

Reply:
Yes. Will it be better if we just invoke `reconfigure` here to enforce that?

Reply:
Err may be not since this might be used specifically just to emit a new block event.